src/main/java/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,34 @@
+package dao;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import model.OperationType;
+
+public class FileReaderImpl implements dao.FileReader {
```

Comment:
The class FileReaderImpl should not begin with an empty line. Please remove the empty line at the start of the class implementation to adhere to the common mistakes checklist.